lib/output/markdown.js

```diff
@@ -1,10 +1,186 @@
 'use strict';
 
 var extend = require('extend'),
-  getTemplate = require('../get_template'),
-  helpers = require('../markdown_helpers'),
-  resolveTheme = require('../resolve_theme'),
-  Handlebars = require('handlebars');
+  mdast = require('mdast'),
+  inlineLex = require('jsdoc-inline-lex');
+
+/**
+ * Format link & tutorial tags with simple code inline tags.
+ *
+ * @param {string} text input - typically a description
+ * @returns {string} markdown-friendly output
+ * @private
+ * @example
+ * formatInlineTags('{@link Foo}'); // "`Foo`"
+ */
+function formatInlineTags(text) {
+  var output = '';
+  var tokens = inlineLex(text);
+
+  for (var i = 0; i < tokens.length; i++) {
+    if (tokens[i].type === 'text') {
+      output += tokens[i].capture[0];
+    } else {
+      output += '`' + tokens[i].capture[1] + '`';
+    }
+  }
+
+  return output;
+}
+
+function formatType(type) {
+  if (!type) {
+    return '';
+  }
+  switch (type.type) {
+  case 'NameExpression':
+    return type.name;
+  case 'UnionType':
+    return type.elements.map(function (element) {
+      return formatType(element);
+    }).join(' or ');
+  case 'AllLiteral':
+    return 'Any';
+  case 'OptionalType':
+    return '[' + formatType(type.expression) + ']';
+  case 'TypeApplication':
+    return formatType(type.expression) + '<' +
+      type.applications.map(function (application) {
+        return formatType(application);
+      }).join(', ') + '>';
+  }
+}
+
+function identity(x) {
+  return x;
+}
+
+function heading(depth, str) {
+  return {
+    type: 'heading',
+    depth: depth,
+    children: [text(str)]
+  };
+}
```

Comment:
You could use something like [unist-builder](https://github.com/eush77/unist-builder) for this if you find you’re repeating yourself too often and prefer a more programmatic take; its like virtual-hyperscript but for the nodes in mdast :smile: